### PR TITLE
DAOS-5859 control: Integrate fault domains with member DB

### PR DESCRIPTION
**Pull request by kjacque**
- Include fault domain in the membership Join request. The
  management service stores the member's parent fault domain
  in the membership database.
- Handle FaultDomain in the JSON marshal/unmarshal methods for
  the membership DB.
- Add member fault domains to system query output.
- Add utility functions to FaultDomain.
- Export the domain level list in the FaultDomain structure.
- Differentiate between nil and empty in FaultDomain.String().

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]
integration looks good

**Review by mjmac — APPROVED** [review]
Looks good. One minor comment, wouldn't block on it.

File: src/control/server/faultdomain_test.go [hunk 1]
```diff
@@ -35,6 +35,16 @@ import (
 	"github.com/daos-stack/daos/src/control/system"
 )
 
+func assertFaultDomainEqualStr(t *testing.T, expResultStr string, result *system.FaultDomain) {
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/control/server/faultdomain_test.go:38]: Ah, good catch. I can fix in the next patch in this series.

**Comment by kjacque** [comment]
Unrelated failures in Functional_Hardware_Medium:
DAOS-5919
DAOS-5873
DAOS-5943

**Comment by kjacque** [comment]
Travis not expected to pass - there was an existing build bug at the time. I've manually built/tested on both Ubuntu and CentOS7 for this branch.

**Pull request merged**